### DISCUSSION TITLE
Toast

### DISCUSSION THREAD

**Discussion by Febakke** [digdir/designsystemet#1822] [Components]
A toast is a small, temporary notification that appears on the screen to provide feedback about an action, system event, or status change without interrupting the user's workflow.

Toasts are typically displayed for a short period of time and then disappear automatically. They are commonly used to confirm that an action was successful, inform users about background processes, or provide non-critical updates.

**Comment by Barsnes** [comment]
Eit par spørsmål rundt `Toast`:
**Skal desse alltid bli lest opp fortløpande for skjermlesar?** 
Det kan bli mange, og være forstyrrande - men ein skjermlesar må få det med seg.

**Skal fokus bli flytta inn i toast når den blir vist?**
Dersom toasten ikkje får fokus kan ikkje dei som bruker tastatur få dei bort, eller trykke på eventuelle handlinga i den.
Ein stor bakdel med at fokus flytta seg, er at brukaren kan være fokusert på eit element som er i aktiv bruk (input, radio, etc.). Dette kan medføre at dei mister kontekst på der dei er, og kan være vanskelig å navigere seg tilbake.
Kva skjer også når ein toast går bort, kor skal fokus?

Eg trur kanskje `Toast` er vanskelig å få laga bra mtp UU. Det er mange fallgruver når det kjem til UU. Det kan være vanskelig for personar med nedsatt synsfelt å sjå den, det kan bli forstyrrande for skjermlesarar dersom dei blir lest opp fortløpande, det er vanskelig å flytte fokus både inn og ut.

**Comment by mrosvik** [comment]
Kom opp på weekly i dag (19.09.25) at vi sammen bør utforme retnignslinjer/mønstre for bruk av Toast. De bør ta for seg: 
- Hva toast er, og hva den brukes til (kort bekreftelse på en handling, ikke-kritisk info)
- Hva den ikke skal brukes til (feil, advarsler, kritisk informasjon)
- Typiske bruksområder (lagring, kopiert til utklippstavle osv.).
- Innhold og tekstlengde
     - Anbefalt kortfattet språk 
     - Ikke lenker eller komplisert formatering
- Varighet / visningstid: Hvor lenge toast skal vises, basert på antall tegn.
- Plassering desktop og mobil
- Lukkes automatisk

**Reply by teaeskeland** [reply]
Ønsker oss veldig toastkomponentet i Designsystemet :)

**Comment by eirikbacker** [comment]
Github just retired their Toast component due to the bad accessibility this pattern brings; https://primer.style/accessibility/toasts/
Should we consider the same? 🤔

**Reply by Barsnes** [reply]
I don't mind, but if we decide to do it I'd like to have an article on why, just like Primer :)

**Comment by Barsnes** [comment]
We now have a dedicated page for `Toast` in our component documentation, but like Primer we opted to fill it with info on why toasts might be a bad idea: https://designsystemet.no/no/components/docs/toast/overview
